### PR TITLE
Split `IYAXSerializer<T>` to `IYAXSerializer<T>`,  `IYAXDeserializer<T>` and `IYAXMultiStateDeserializer<T>`

### PR DESCRIPTION
**Pull request by 304NotModified**
Added also `IYAXMultiStateDeserializer<T>` otherswise `IYAXDeserializer<T>` can't be covariant

### CODE REVIEW

File: YAXLib/IYAXMultiStateDeserializer.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (C) Sina Iravanian, Julian Verdurmen, axuno gGmbH and other contributors.
+// Licensed under the MIT license.
+
+namespace YAXLib
+{
+    public interface IYAXMultiStateDeserializer<T> : IYAXDeserializer<T>
```
Comment (304NotModified) [inline YAXLib/IYAXMultiStateDeserializer.cs:6]: Otherswise IYAXDeserializer<T> can't be covariant

**Comment by axunonb** [comment]
Really an excellent way to distinguish

**Comment by axunonb** [comment]
Unavoidable, but only 2 conflicts easy to resolve ;-)

**Comment by 304NotModified** [comment]
I'm sorry, busy weeks. Will try to work on the pull requests this weekend

**Comment by 304NotModified** [comment]
Will reopen when I have more time

**Pull request closed without merging**